File: c#/Structural/Decorator/Models/CustomerMessage.cs
using Interface.IMessage;

namespace Models.CustomerMessage
{
    public class CustomerMessage : IMessage
    {
        protected string name;
        protected string email;
        protected int telephone;

        public CustomerMessage(string name, string email, int telephone)
        {
            this.name = name;
            this.email = email;
            this.telephone = telephone;
        }

        public override string ToString()
        {
            return $"Name {this.name} Email: {this.email} Telephone: {this.telephone}";
        }

        public IMessage ProcessMessage()
        {
            return this;
        }

        public string GetContent()
        {
            return this.ToString();
        }

        public void SetContent(string content)
        {
            throw new Exception("Not supported yet.");
        }
    }
}

File: typescript/src/structural/decorator/models/CustomerMessage.ts
import { IMessage } from "../interfaces/IMessage";

export class CustomerMessage implements IMessage {
    private name: string;
    private email: string;
    private telephone: number;

    constructor(name: string, email: string, telephone: number){
        this.name = name
        this.email = email
        this.telephone = telephone
    }

    processMessage(): IMessage {
        return this
    }

    toString(): string {
        return `Name: ${this.name} Email: ${this.email} Telephone: ${this.telephone}`
    }

    getContent(): string {
        return this.toString();
    }

    setContent(content: string): void {
        throw new Error("Not supported yet.")
    }

}